src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/wataskconfigurationapi/controllers/TaskConfigurationControllerTest.java

```diff
@@ -362,17 +365,45 @@ void should_succeed_and_return_configuration_with_new_role_name_case_manager() t
             .andReturn();
     }
 
+    @DisplayName("Should return 502 when ccd data store api is down")
+    @Test
+    void should_return_status_code_502_when_ccd_data_store_api_is_down() throws Exception {
+
+        setupRoleAssignmentResponse(true);
+        configure3rdPartyResponses();
+
+        doThrow(FeignException.BadRequest.class)
+            .when(ccdDataServiceApi)
+            .getCase(any(), any(), any());
+
+        mockMvc.perform(
+                post(TASK_CONFIGURATION_ENDPOINT + testTaskId)
+                    .contentType(APPLICATION_JSON_VALUE)
+            )
+            .andExpectAll(
+                status().is5xxServerError(),
```

Comment:
Should this not check that it is a 502?

Reply:
This method checks error code starts with 5 or not.
Here is their explanation 'Assert the response status code is in the 5xx range'
Should I take any action?

Reply:
This method checks error code starts with 5 or not.
Here is their explanation 'Assert the response status code is in the 5xx range'
Should I take any action?

Reply:
My point is that you are checking that it is any 5xx error when it fact your test should be checking it is specifically a 502 status code?  
If someone later changes the status to 503 or 500 then this test would still pass but it's not correct.